src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -488,6 +489,18 @@ private void ParseVulnerability(IResultLogWriter output)
                 {
                     ParseLocationsFromTraces(result);
                 }
+                else if (AtStartOfNonEmpty(_strings.DefaultSeverity))
+                {
+                    rule.DefaultConfiguration.SetProperty("Deprecated"+ _strings.DefaultSeverity, _reader.ReadElementContentAsString());
+                }
+                else if (AtStartOfNonEmpty(_strings.InstanceSeverity))
+                {
+                    rule.SetProperty("Deprecated"+ _strings.InstanceSeverity, _reader.ReadElementContentAsString());
```

Comment:
>Deprecated [](start = 38, length = 10)

Why are we prefixing these properties with 'Deprecated'? they never appeared previously, did they? If these are deprecated values, where are the current ones? :)

Reply:
We want to tell people who are considering using these Fortify-specific property bag values in their analysis that Fortify itself has deprecated them.

---
In reply to: [315392545](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315392545) [](ancestors = 315392545)

Reply:
If they are deprecated, why are we populating them? I'm not sure I like the idea of the converter adding this annotation to the property name. Are these values going away entirely? If they are deprecated, doesn't that mean that our severity construction mechanism is based on deprecated data?

---
In reply to: [315393434](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315393434) [](ancestors = 315393434,315392545)

Reply:
The answer to your last question is "no." Our `level` calculation is based purely on Fortify's `Impact`. According to Yekaterina, the calculation should also include `Probability`, but that doesn't occur in the FPR file.

We are populating them because consumers of Fortify FPR files might be using these Fortify-specific values, and SARIF has no place for them.

If you object to our editorializing in the names, I don't object to removing the "Deprecated". Harleen/Surbhi and I discussed and felt it was a good idea, but I don't think any of us is going to fight for it if you object.

---
In reply to: [315401208](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315401208) [](ancestors = 315401208,315393434,315392545)

Reply:
+1. to add to what Larry explained:  this requirement originated when Craig Gehre (CSE) asked specifically for the missing "severity".

The reason for appending "deprecated" to it is because MicroFocus confirmed that it is so. Having this property here, named thusly gives us the benefit of both satisfying our customer's need while ensuring they know this is a deprecated property.

---
In reply to: [315405175](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315405175) [](ancestors = 315405175,315401208,315393434,315392545)

Reply:
(i also dont object to removing "deprecated" word from there, though i'd prefer it :)

---
In reply to: [315456813](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315456813) [](ancestors = 315456813,315405175,315401208,315393434,315392545)

Reply:
Please remove the term 'Deprecated' here and elsewhere. Remember the driving design principle of a converter is to introduce as little special transformation itself. The ideal converter moves relevant data into existing SARIF slots and otherwise persists other data in an appropriate property bag. We do not transform property names, in order to facilitate a user's ability to correlate a property to a data value in the original log file. All of this helps reduce compatibility issues related to the converter code.

The Fortify log file format is owned by MicroFocus. Imagine that in three months, they decide to undeprecate DefaultSeverity. They have every right to do so. If in this time-frame, pipelines have produced SARIF with 'DeprecatedDefaultSeverity', then consumers will have taken a dependency on this property name. IE, we have created a compatibility issue based on our own decisions. 

As a second point, introducing a prefix such as 'Deprecated' raises a point of consistency: does this mean that every other property value is verifiably not deprecated? I don't believe we've made that assessment.

So please just flow the MicroFocus property name forward., Note that we retain the casing of these property names (despite SARIF generally advising camel casing) for the same 'hands off' converter principle.



---
In reply to: [315456930](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315456930) [](ancestors = 315456930,315456813,315405175,315401208,315393434,315392545)

Reply:
+1

---
In reply to: [315777514](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r315777514) [](ancestors = 315777514,315456930,315456813,315405175,315401208,315393434,315392545)